jdaviz/utils.py

```diff
@@ -286,31 +288,48 @@ def standardize_metadata(metadata):
     return out_meta
 
 
-def flux_conversion(spec, values, original_units, target_units):
+def indirect_units():
+    return [
+        u.erg / (u.s * u.cm**2 * u.Angstrom * u.sr),
+        u.erg / (u.s * u.cm**2 * u.Hz * u.sr),
+        u.ph / (u.Angstrom * u.s * u.cm**2 * u.sr), u.ph / (u.Angstrom * u.s * u.sr * u.cm**2),
+        u.ph / (u.s * u.cm**2 * u.Hz * u.sr)
+    ]
+
+
+def flux_conversion(values, original_units, target_units, spec=None, eqv=None, slice=None):
     """
-    Given a Spectrum1D object, flux values, original flux units, and target units,
-    this method will return the flux values in the converted units. This conversion
-    takes into account the possible surface brightness to flux or vice versa change that
-    may happen between units.
+    Convert flux or surface brightness values from original units to target units.
+
+    This function handles the conversion of flux or surface brightness values between different
+    units, taking into account changes between flux and surface brightness. It supports complex
+    conversions for Spectrum1D objects or cube image data.
 
     Parameters
     ----------
-    spec : `~specutils.Spectrum1D` object
-        The Spectrum1D object that will have converted flux units.
-
     values : float array
-        Flux values in the original units.
+        Flux or surface brightness values in the original units.
 
     original_units : str
-        The flux units of the spec object.
+        The flux or surface brightness units of the spec object or cube image.
 
     target_units : str
-        The units the flux will be converted to.
+        The units the flux or surface brightness will be converted to.
+
+    spec : `~specutils.Spectrum1D`, optional
+        The Spectrum1D object that will have converted flux or surface brightness units.
+
+    eqv : list of `astropy.units.equivalencies`, optional
+        A list of Astropy equivalencies necessary for complex unit conversions/translations.
+
+    slice : `astropy.units.Quantity`, optional
+        The current slice of a data cube, with units. Necessary for complex unit
+        conversions/translations that require spectral density equivalencies.
```

Comment:
could this be unitless and just documented as required to be in the same units as `original_units`/`values`?  That would then save a call to `app._get_display_units` when calling this.